packages/wallets/src/features/cashier/modules/Transfer/components/TransferFormDropdown/TransferFormDropdown.tsx

```diff
@@ -62,38 +62,19 @@ const TransferFormDropdown: React.FC<TProps> = ({ fieldName, mobileAccountsListR
         (account: TInitialTransferFormValues['fromAccount']) => {
             if (account?.loginid === selectedAccount?.loginid) return;
 
-            const swapAccounts = () => {
+            if (isFromAccountDropdown) {
                 setValues(prev => {
+                    const toAccount = account?.loginid !== activeWallet?.loginid ? activeWallet : undefined;
+
                     return {
                         ...prev,
                         activeAmountFieldName: undefined,
-                        fromAccount: isFromAccountDropdown ? account : prev.toAccount,
-                        fromAmount: prev.toAmount,
-                        toAccount: isFromAccountDropdown ? prev.fromAccount : account,
-                        toAmount: prev.fromAmount,
+                        fromAccount: account,
+                        fromAmount: 0,
+                        toAccount,
+                        toAmount: 0,
                     };
                 });
-            };
-
-            if (isFromAccountDropdown) {
```

Comment:
love to see code removed :D though also wondering a bit why it was there to begin with :D